### PR TITLE
Remove integration of Delayed Job

### PR DESCRIPTION
**Pull request by jeroenvandijk**
Currently email_spec is working of jobs when it has detected Delayed Job. I think this is really undesirable. 

* First it causes unexpected behavior. I've once spent some hours to detect why my scenario didn't seem to populate any jobs. After a while I figured that email-spec had already worked them off. 
* Secondly it is unnecessary. I spent a long time debugging when my test suite run time exploded after adding a particular scenario with a lot of objects being created. My test suites run time went from 3 minutes to 8 minutes. The reason was that all those object had after save callbacks with background jobs. This meant that there was a big queue of background jobs. For this particular scenario they didn't have to be run, but email-spec triggered the worker anyways causing a big extra delay.

So in summary I think the integration of delayed job should be removed. If it is necessary for your scenario to pass, it is better to explicitly call the worker yourself IMHO.

WDYT?

Jeroen

### CODE REVIEW

**Comment by etagwerker** [comment]
I agree with @jeroenvandijk and @doolin. I don't think it's necessary inside the `email-spec` gem. If we have `DelayedJob` in, then it means we should also have `Sidekiq`, `Resque` and any other background job gem. 

This means that the gem will continue to grow in size for something that is not core to the goal of this library. 

Also, it can produce unexpected behavior when you call `last_email_sent` as it works off DJ, without considering if there are other jobs in the queue. https://github.com/bmabey/email-spec/blob/master/lib/email_spec/background_processes.rb#L9

IMO programmers should handle their queues however they want to, without any interference by `email-spec`

Finally, I think that [this method](https://github.com/bmabey/email-spec/blob/master/lib/email_spec/background_processes.rb#L28-L37) assumes too much and that [these methods](https://github.com/bmabey/email-spec/blob/master/lib/email_spec/background_processes.rb#L4-L22) go against the [principle of least astonishment](http://c2.com/cgi/wiki?PrincipleOfLeastAstonishment)
